test-crawler/extractor/extractor.go

```diff
@@ -204,17 +232,41 @@ func getFunctionNodes(content string, treeCursor *sitter.TreeCursor, parser *a.P
 				}
 
 				funcAnnoPair = append(funcAnnoPair, struct {
```

Comment:
We are redefining this struct here and in the function return values declaration.
Therefore, it is a good idea to declare the struct outside the function: we will avoid code duplication, and make the code more readable.